src/status_im2/contexts/quo_preview/share/qr_code.cljs

```diff
@@ -1,55 +1,97 @@
+               :value "None"}]}
+   (preview/customization-color-option)
+   {:key  :full-name
+    :type :text}])
+
+(def wallet-account-descriptor
+  [{:key     :emoji
+    :type    :select
+    :options [{:key "🍒"}
+              {:key "🐧"}
+              {:key "🍨"}]}
+   (preview/customization-color-option)])
+
+(def channel-descriptor
+  [{:key     :emoji
+    :type    :select
+    :options [{:key "🍒"}
+              {:key "🐧"}
+              {:key "🍨"}]}
+   (preview/customization-color-option)])
+
+(def saved-address-descriptor
+  [{:key  :f-name
+    :type :text}
+   {:key  :l-name
+    :type :text}
+   (preview/customization-color-option)])
 
-(defn preview-qr-code
+(defn preview
   []
-  (let [state                  (reagent/atom {:text                   "https://status.im"
-                                              :error-correction-level :highest})
-        text                   (reagent/cursor state [:text])
-        error-correction-level (reagent/cursor state [:error-correction-level])
-        media-server-uri       (reagent/atom "")]
+  (let [state (reagent/atom
+               {:url                 "https://join.status.im/status"
+                :media-server-port   (rf/sub [:mediaserver/port])
+                :size                250
+                :avatar              :none
+                :profile-picture     (resources/get-mock-image :user-picture-male5)
+                :customization-color :army
+                :full-name           "Full Name"
+                :emoji               "🍒"
+                :picture             (resources/get-mock-image :community-logo)
+                :f-name              "First Name"
+                :l-name              "Last Name"})]
     (fn []
-      (reset! media-server-uri (image-server/get-qr-image-uri-for-any-url
-                                {:url         @text
-                                 :port        (rf/sub [:mediaserver/port])
-                                 :error-level @error-correction-level
-                                 :qr-size     250}))
       [preview/preview-container
        {:state      state
-        :descriptor descriptor}
-       [rn/view {:style {:padding-bottom 150}}
-        [rn/view {:style {:flex 1}}]
-        [rn/view
-         {:style {:padding-vertical 60
-                  :flex-direction   :row
-                  :justify-content  :center}}
-         [rn/view
-          [quo/qr-code
-           {:source {:uri @media-server-uri}
-            :height 250
-            :width  250}]
+        :descriptor (concat descriptor
+                            (case (:avatar @state)
+                              :profile        profile-descriptor
+                              :wallet-account wallet-account-descriptor
+                              :channel        channel-descriptor
+                              :saved-address  saved-address-descriptor
+                              nil))}
+       [rn/view
+        {:style {:flex            1
```

Comment:
FYI: You can pass this style map to `preview-container` in the prop `component-container-style` and then remove the parent `view`.

Reply:
Done